### DISCUSSION TITLE
Symbols for Query Keys

### DISCUSSION THREAD

**Discussion by Miteshdv** [TanStack/query#395] [General]
I have one question why not Javascript symbols instead of String for Query Key

**Comment by wolverineks** [comment]
what are the benefits of symbols?

**Reply by joaopedrodcf** [reply]
Symbols prevent collisions

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol

![image](https://github.com/TanStack/query/assets/11476152/d20f7eb9-e0f2-4acc-bcb2-7270be1eb94d)

**Reply by TkDodo** [reply]
this is a super old discussion, but the bottom line is: symbols aren't json serializable
